### PR TITLE
DATAUP-330 final push

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes
* Found that a number of jquery event listeners that get bound to $(document), don't go away as expected. They should, now, before other tests call them - fixes the page reload issues.
* Found that a number of timeouts set by widgets are still function after running tests, which forces the widgets to stay in memory and execute code after tests run, if they run long enough. This fixes the Datalist issues, and a few others. There's likely more still lurking.
* Fixed issues with `require.js` timing out in the `narrativeViewers` and `kbaseNarrativeOutputCell` tests. In `narrativeViewers`, this was fixed by mocking `window.require`. `kbaseNarrativeOutputCell` will need some more work, as mocking `require` didn't do the fix - that particular test was commented out instead and might require some refactoring.
* The `narrativeViewers` fix was accompanied by a little refactor.
* Other places weren't properly clearing their impacts on the DOM after running tests. The `jobCommChannel` error dialog test (and code) now gets around this by putting a very specific selector on the OK button, just in case other tests leave dialogs in the DOM - this was one cause of the page reload issue.
* Refactored `kbaseNarrativeSpec.js` to be less weird about Promise management in the tests, also removes jquery listeners after tests.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-330
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
Run the tests!
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeDataList.js [hunk 1]
```diff
@@ -305,6 +305,7 @@ define([
 
             // listener for refresh
             $(document).on('updateDataList.Narrative', () => {
+                console.log('updating datalist from trigger');
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeDataList.js:308]: should this be left in?

File: test/unit/spec/kbaseNarrativeSpec.js [hunk 2]
```diff
@@ -97,51 +104,50 @@ define([
...
+            // clear all jquery event listeners set up by either NarrativeLogin or anything else
+            $(document).off();
+            $([Jupyter.events]).off();
+            document.body.innerHTML = '';
         });
 
         it('Should instantiate', () => {
             const narr = new Narrative();
             expect(narr.maxNarrativeSize).toBe('10 MB');
+            narr.loadingWidget.clearTimeout();
```
Comment (ialarmedalien) [inline test/unit/spec/kbaseNarrativeSpec.js:118]: if you move creating the new narrative to the `beforeEach`, you can put `narr.loadingWidget.clearTimeout();` in the `afterEach`.

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeDataList.js (hunk 1 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeDataList.js:308]: Nope! Thanks.

File: test/unit/spec/kbaseNarrativeSpec.js (hunk 2 above)
Comment (briehl) [reply test/unit/spec/kbaseNarrativeSpec.js]: Good idea. Fixed.

File: test/unit/spec/narrative_core/narrativeViewersSpec.js [hunk 3]
```diff
@@ -119,14 +119,39 @@ define(['narrativeViewers', 'narrativeConfig'], (Viewers, Config) => {
...
+        });
+
         it('should create a viewer, given inputs', () => {
             const dataCell = {
                 obj_info: {
                     type: 'Module.Type1-1.0',
                     bare_type: 'Module.Type1',
                 },
             };
+            // spyOn(window, 'require').and.returnValue(new Error('nope'));
```
Comment (ialarmedalien) [inline test/unit/spec/narrative_core/narrativeViewersSpec.js:147]: comment accidentally left in?

**Review by ialarmedalien — APPROVED** [review]
👍

File: test/unit/spec/narrative_core/narrativeViewersSpec.js (hunk 3 above)
Comment (briehl) [reply test/unit/spec/narrative_core/narrativeViewersSpec.js:147]: So much easier to remember these in the git diff viewer...

**Pull request merged**